logdb/logdb.go

```diff
@@ -104,143 +105,174 @@ FROM (%v) e
 	LEFT JOIN ref r5 ON e.topic1 = r5.id
 	LEFT JOIN ref r6 ON e.topic2 = r6.id
 	LEFT JOIN ref r7 ON e.topic3 = r7.id
-	LEFT JOIN ref r8 ON e.topic4 = r8.id`
+	LEFT JOIN ref r8 ON e.topic4 = r8.id
+	%v
+	`
 
 	if filter == nil {
-		return db.queryEvents(ctx, fmt.Sprintf(query, "event"))
+		to, err := newSequence(MaxBlockNumber, 0, 0)
+		if err != nil {
+			return nil, err
+		}
+		where := fmt.Sprintf(" WHERE e.seq >= 0 AND e.seq <= %v", to)
+		return db.queryEvents(ctx, fmt.Sprintf(query, where))
 	}
 
 	var (
-		subQuery = "SELECT seq FROM event WHERE 1"
-		args     []any
+		whereOrderLimit strings.Builder
+		args            []any
 	)
 
 	if filter.Range != nil {
-		subQuery += " AND seq >= ?"
+		whereOrderLimit.WriteString(" WHERE seq >= ?")
 		from, err := newSequence(filter.Range.From, 0, 0)
 		if err != nil {
 			return nil, err
 		}
 		args = append(args, from)
 		if filter.Range.To >= filter.Range.From {
-			subQuery += " AND seq <= ?"
+			whereOrderLimit.WriteString(" AND seq <= ?")
 			to, err := newSequence(filter.Range.To, txIndexMask, logIndexMask)
 			if err != nil {
 				return nil, err
 			}
 			args = append(args, to)
 		}
+	} else {
+		to, err := newSequence(MaxBlockNumber, 0, 0)
+		if err != nil {
+			return nil, err
+		}
+		whereOrderLimit.WriteString(fmt.Sprintf(" WHERE e.seq > 0 AND e.seq <= %v", to))
```

Comment:
```suggestion
		whereOrderLimit.WriteString(fmt.Sprintf(" WHERE e.seq >= 0 AND e.seq <= %v", to))
```

Genesis block can contain event and tx/log(Index) are starting from 0.

Reply:
Cheers added!